src/storage/src/table/cell_based_table.rs

```diff
@@ -367,7 +381,134 @@ impl<S: StateStore> CellBasedTableRowIter<S> {
         Ok(iter)
     }
 
-    pub async fn collect_data_chunk(
+    async fn next_pk_and_row(&mut self) -> StorageResult<Option<(Vec<u8>, Row)>> {
+        loop {
+            match self.iter.next().await? {
+                None => return Ok(self.cell_based_row_deserializer.take()),
+                Some((key, value)) => {
+                    tracing::trace!(
+                        target: "events::storage::CellBasedTable::scan",
+                        "CellBasedTable scanned key = {:?}, value = {:?}",
+                        key,
+                        value
+                    );
+                    let pk_and_row = self
+                        .cell_based_row_deserializer
+                        .deserialize(&key, &value)
+                        .map_err(err)?;
+                    match pk_and_row {
+                        Some(_) => return Ok(pk_and_row),
+                        None => {}
+                    }
+                }
+            }
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl<S: StateStore> TableIter for CellBasedTableRowIter<S> {
+    async fn next(&mut self) -> StorageResult<Option<Row>> {
+        self.next_pk_and_row()
+            .await
+            .map(|r| r.map(|(_pk, row)| row))
+    }
+}
+
+#[async_trait::async_trait]
+impl<S: StateStore> CellTableChunkIter for CellBasedTableRowIter<S> {}
```

Comment:
After the whole feature (encoding part also) is done, I guess `CellBasedTableRowIter` will never be used as returning a `DataChunk` probably? 🤔 

And this `CellTableChunkIter` will only be used by `DedupPkCellBasedTableRowIter`. Maybe we can save one trait, we can revisit this in the future.

Reply:
Agree with this